cirq/experiments/random_quantum_circuit_generation.py

```diff
@@ -289,16 +294,68 @@ def _get_active_pairs(graph: nx.Graph, grid_layer: GridInteractionLayer):
 
 @dataclasses.dataclass(frozen=True)
 class CircuitLibraryCombination:
-    """For a given layer (which defines a set of pairs of qubits), `combinations` is a 2d array
+    """For a given layer (specifically, a set of pairs of qubits), `combinations` is a 2d array
     of shape (n_combinations, len(pairs)) where each row represents a combination (with replacement)
     of two-qubit circuits. The actual values are indices into a list of library circuits.
 
-    `layer` is used for record-keeping but is otherwise unnecessary. Therefore, it is optional.
+    `layer` is used for record-keeping. This is the GridInteractionLayer if using
+    `get_random_combinations_for_device`, the Moment if using
+    `get_random_combinations_for_layer_circuit` and ommitted if using
+    `get_random_combinations_for_pairs`.
     """
 
-    layer: Optional[GridInteractionLayer]
+    layer: Optional[Any]
     combinations: np.array
-    pairs: List[Tuple['cirq.Qid', 'cirq.Qid']]
+    pairs: List[QidPairT]
+
+
+def _get_random_combinations(
+    n_library_circuits: int,
+    n_combinations: int,
+    *,
+    pair_gen: Generator[Tuple[List[QidPairT], Any], None, None],
```

Comment:
Why limit yourself to Generator? Iterable is broader and applicable to your code as well.

Reply:
I did this suggestion. Don't know why my reply didn't post